lib/versions.ps1

```diff
@@ -1,53 +1,231 @@
+        [String]
+        $App,
+        [Parameter(Position = 1)]
+        [String]
+        $Bucket,
+        [Parameter(Position = 2)]
+        [String]
+        $URL
+    )
+    return (manifest $App $Bucket $URL).version
 }
-function versions($app, $global) {
-    $appdir = appdir $app $global
-    if(!(test-path $appdir)) { return @() }
 
-    sort_versions (Get-ChildItem $appdir -dir -attr !reparsePoint | Where-Object { $null -ne $(Get-ChildItem $_.fullname) } | ForEach-Object { $_.name })
+function Select-CurrentVersion {
+    <#
+    .SYNOPSIS
+        Select current version of app
+    .DESCRIPTION
+        Select current version of installed app, from 'current\manifest.json' or modified time of version directory
+    .PARAMETER App
+        App's name
+    .PARAMETER Global
+        Globally installed application
+    #>
+    [OutputType([String])]
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $true, Position = 0, ValueFromPipeline = $true)]
+        [String]
+        $App,
+        [Parameter(Position = 1)]
+        [Switch]
+        $Global
+    )
+
+    $appPath = appdir $App $Global
+    if (Test-Path "$appPath\current") {
+        $currentVersion = (installed_manifest $App 'current' $Global).version
+    } else {
+        $installedVersion = Get-InstalledVersion -App $App -Global:$Global
+        if ($installedVersion) {
+            $currentVersion = $installedVersion[-1]
+        } else {
+            $currentVersion = $null
+        }
+    }
+    return $currentVersion
 }
 
-function version($ver) {
-    $ver -split '[\.-]' | ForEach-Object {
-        $num = $_ -as [int]
-        if($num) { $num } else { $_ }
+function Get-InstalledVersion {
+    <#
+    .SYNOPSIS
+        Get installed version of app
+    .DESCRIPTION
+        Get all installed version of app, by checking version directories' 'install.json'
+    .PARAMETER App
+        App's name
+    .PARAMETER Global
+        Globally installed application
+    #>
+    [OutputType([Object[]])]
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $true, Position = 0, ValueFromPipeline = $true)]
+        [String]
+        $App,
+        [Parameter(Position = 1)]
+        [Switch]
+        $Global
+    )
+
+    $appPath = appdir $App $Global
+    if (!(Test-Path $appPath)) {
+        return @()
+    } else {
+        return @((Get-ChildItem "$appPath\*\install.json" | Sort-Object -Property LastWriteTimeUtc).Directory.Name) -ne 'current'
     }
+
+    # Deprecated
+    # sort_versions (Get-ChildItem $appPath -dir -attr !reparsePoint | Where-Object { $null -ne $(Get-ChildItem $_.FullName) } | ForEach-Object { $_.Name })
 }
-function compare_versions($a, $b) {
-    $ver_a = @(version $a)
-    $ver_b = @(version $b)
 
-    for($i=0;$i -lt $ver_a.length;$i++) {
-        if($i -gt $ver_b.length) { return 1; }
+function Compare-Version {
+    <#
+    .SYNOPSIS
+        Compare versions
+    .DESCRIPTION
```

Comment:
No need to specify description when it differ only with few words. Description is detailed version of brief synopsis. There is no reason that synopsis could not contain `Compare versions, mainly according to SemVer's rules`

Same for other functions